examples/step-37/doc/results.dox

```diff
@@ -11,122 +11,193 @@ solution through both isocontours and volume rendering:
+@endcode
+
+Since $\mathcal Q_4$ elements on a certain mesh correspond to $\mathcal Q_2$
+elements on half the mesh size, we can compare the run time at cycle 4 with
+fourth degree polynomials with cycle 5 using quadratic polynomials, both at
+2.1 million degrees of freedom. The surprising effect is that the solver for
+$\mathcal Q_4$ element is actually slightly faster than for the quadratic
+case, despite using one more linear iteration. The effect that higher-degree
+polynomials are similarly fast or even faster than lower degree ones is one of
+the main strengths of matrix-free operator evaluation through sum
+factorization, see the <a
+href="http://dx.doi.org/10.1016/j.compfluid.2012.04.012">matrix-free
+paper</a>. This is fundamentally different to matrix-based methods that get
+more expensive per unknown as the polynomial degree increases and the coupling
+gets denser.
+
+In addition, also the setup gets a bit cheaper for higher order, which is
+because fewer elements need to be set up.
+
+Finally, let us look at the timings with degree 8, which corresponds to
+another round of mesh refinement in the lower order methods:
+
+@code
+Cycle 0
+Number of degrees of freedom: 4913
+Total setup time               (wall) 0.0842004s
+Time solve (8 iterations)  (CPU/wall) 0.019296s/0.0192959s
+
+Cycle 1
+Number of degrees of freedom: 35937
+Total setup time               (wall) 0.327048s
+Time solve (8 iterations)  (CPU/wall) 0.07517s/0.075999s
+
+Cycle 2
+Number of degrees of freedom: 274625
+Total setup time               (wall) 2.12335s
+Time solve (8 iterations)  (CPU/wall) 0.448739s/0.453698s
+
+Cycle 3
+Number of degrees of freedom: 2146689
+Total setup time               (wall) 16.1743s
+Time solve (8 iterations)  (CPU/wall) 3.95003s/3.97717s
+
+Cycle 4
+Number of degrees of freedom: 16974593
+Total setup time               (wall) 130.8s
+Time solve (8 iterations)  (CPU/wall) 31.0316s/31.767s
+@endcode
+
+Here, the initialization seems considerably slower than before, which is
+mainly due to the computation of the diagonal of the matrix, which actually
+computes a 729 x 729 matrix on each cell and throws away everything but the
+diagonal. The solver times, however, are again very close to the quartic case,
+showing that the linear increase with the polynomial degree that is
+theoretically expected is almost completely offset by better computational
+characteristics and the fact that higher order methods have a smaller share of
+degrees of freedom living on several cells that add to the evaluation
+complexity.
 
 <h3>Comparison with a sparse matrix</h3>
 
 In order to understand the capabilities of the matrix-free implementation, we
-compare the performance on the 3d example above with a SparseMatrix
-implementation and we measure the computation times for both initialization of
+compare the performance of the 3d example above with a SparseMatrix
+implementation by measuring both the computation times for the initialization of
```

Comment:
I re-tested this now. I used SparseMatrix before but now use a Trilinos matrix with MPI. I adjusted the numbers (they are quite close for the matrix-based scheme, even though I use a newer laptop processor now - the matrix-free code improved considerably more).